From b297329d12abcf1347b0bd55a1550c8758cad246 Mon Sep 17 00:00:00 2001
From: Matthew Waters <matthew@centricular.com>
Date: Tue, 16 Oct 2018 19:01:27 +1100
Subject: [PATCH] meson: fix HAVE_INIT_ONCE declaration

meson will set the HAVE_INIT_ONCE value to 0 or 1 when using
conf.set10() so an #if defined() will always succeed.
---
 src/graphene-box.c     | 2 +-
 src/graphene-private.h | 2 +-
 src/graphene-vectors.c | 6 +++---
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/graphene-box.c b/src/graphene-box.c
index 489f378..a5995d8 100644
--- a/src/graphene-box.c
+++ b/src/graphene-box.c
@@ -677,7 +677,7 @@ init_static_box (void)
     }
 }
 
-#elif defined(HAVE_INIT_ONCE)
+#elif defined(HAVE_INIT_ONCE) && HAVE_INIT_ONCE
 static INIT_ONCE static_box_once = INIT_ONCE_STATIC_INIT;
 
 static BOOL CALLBACK
diff --git a/src/graphene-private.h b/src/graphene-private.h
index e8f9098..cdf016b 100644
--- a/src/graphene-private.h
+++ b/src/graphene-private.h
@@ -25,7 +25,7 @@
 #define __GRAPHENE_PRIVATE_H__
 
 #include "config.h"
-#ifdef HAVE_INIT_ONCE
+#if defined(HAVE_INIT_ONCE) && HAVE_INIT_ONCE
 #define _WIN32_WINNT 0x0600
 #include <windows.h>
 #endif
diff --git a/src/graphene-vectors.c b/src/graphene-vectors.c
index 7701b3d..d81463a 100644
--- a/src/graphene-vectors.c
+++ b/src/graphene-vectors.c
@@ -496,7 +496,7 @@ init_static_vec2 (void)
     }
 }
 
-#elif defined(HAVE_INIT_ONCE)
+#elif defined(HAVE_INIT_ONCE) && HAVE_INIT_ONCE
 static INIT_ONCE static_vec2_once = INIT_ONCE_STATIC_INIT;
 
 static BOOL CALLBACK
@@ -1177,7 +1177,7 @@ init_static_vec3 (void)
     }
 }
 
-#elif defined(HAVE_INIT_ONCE)
+#elif defined(HAVE_INIT_ONCE) && HAVE_INIT_ONCE
 static INIT_ONCE static_vec3_once = INIT_ONCE_STATIC_INIT;
 
 static BOOL CALLBACK
@@ -1876,7 +1876,7 @@ init_static_vec4 (void)
     }
 }
 
-#elif defined(HAVE_INIT_ONCE)
+#elif defined(HAVE_INIT_ONCE) && HAVE_INIT_ONCE
 static INIT_ONCE static_vec4_once = INIT_ONCE_STATIC_INIT;
 
 static BOOL CALLBACK
-- 
2.19.0

